coderd/httpmw/apikey_test.go

```diff
@@ -991,3 +991,237 @@ func TestAPIKey(t *testing.T) {
 		require.Equal(t, http.StatusOK, res.StatusCode)
 	})
 }
+
+func TestAPIKeyExpiryRefresh(t *testing.T) {
+	t.Parallel()
+
+	t.Run("ShortLivedKeyRefresh", func(t *testing.T) {
+		t.Parallel()
+		var (
+			db, _ = dbtestutil.NewDB(t)
+			user  = dbgen.User(t, db, database.User{})
+			now   = time.Now()
+			// 2 minute lifetime
+			lifetime = int64(120)
+			// Key expires in 30 seconds (less than half the 2-minute lifetime)
+			sentAPIKey, token = dbgen.APIKey(t, db, database.APIKey{
+				UserID:          user.ID,
+				LastUsed:        dbtime.Now().Add(-time.Hour),
+				ExpiresAt:       dbtime.Time(now.Add(30 * time.Second)),
+				LifetimeSeconds: lifetime,
+			})
+			r  = httptest.NewRequest("GET", "/", nil)
+			rw = httptest.NewRecorder()
+		)
+		r.Header.Set(codersdk.SessionTokenHeader, token)
+
+		// Mock time to be exactly now
+		successHandler := http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+			httpapi.Write(r.Context(), rw, http.StatusOK, codersdk.Response{
+				Message: "It worked!",
+			})
+		})
+
+		httpmw.ExtractAPIKeyMW(httpmw.ExtractAPIKeyConfig{
+			DB:              db,
+			RedirectToLogin: false,
+		})(successHandler).ServeHTTP(rw, r)
+		res := rw.Result()
+		defer res.Body.Close()
+		require.Equal(t, http.StatusOK, res.StatusCode)
```

Comment:
Could we just make another API call instead of mocking all this?  Fetch `client.User` or something.